pkg/controllers/consolidation/controller.go

```diff
@@ -116,6 +117,8 @@ func (c *Controller) run(ctx context.Context) {
 type candidateNode struct {
 	*v1.Node
 	instanceType   cloudprovider.InstanceType
+	capacityType   string
```

Comment:
nit: Consider just relying on node.Labels instead of hoisting them up onto this struct.

Reply:
Out of curiosity what's the concern with having this data in the struct since we need to have this data as part of the consolidation savings calculation and we are relying on things like capacity type in other places in the consolidation code like for checking spot -> spot consolidation

Reply:
It's completely optional. Generally, I try to deduplicate as many concepts as possible, and just rely on a few well known concepts. Nodes naturally have a set of properties expressed in their labels, and if I was writing code in some function and had a node, I wouldn't hesitate to use those labels unless I couldn't for some reason. Since candidateNode is itself a node, you're more or less guaranteed to always have those properties available wherever you need them.